TSPL.docc/LanguageGuide/ErrorHandling.md

```diff
@@ -699,6 +699,204 @@ let photo = try! loadImage(atPath: "./Resources/John Appleseed.jpg")
   ```
 -->
 
+## Specifying the Error Type
+
+All of the examples above use the most common kind of error handling,
+where the errors that your code throws
+can be values of any type that conforms to the `Error` protocol.
+This approach matches the reality that
+you don't know ahead of time every error that could happen
+while the code is running,
+especially when propagating errors thrown somewhere else.
+This approach also reflects the fact that errors can change over time.
+New versions of a library ---
+including libraries that your dependencies use ---
+can throw new errors,
+and the rich complexity of real-world user configurations
+can expose failure modes that weren't visible during development or testing.
+The error handling code in the examples above
+always includes a default case to handle errors
+that don't have a specific `catch` clause.
+
+Most Swift code doesn't specify the type for the errors it throws.
+However,
+you might limit code to throwing errors of only one specific type
+in the following special cases:
+
+- When running code on an embedded system
+  that doesn't support dynamic allocation of memory.
+  Throwing an instance of `any Error` or another boxed protocol type
+  requires allocating memory at runtime to store the error.
+  In contrast,
+  throwing an error of a specific type
+  lets Swift allocate that memory upfront.
+
+- When the errors are an implementation detail of some unit of code,
+  like a library,
+  and aren't part of the interface to that code.
+  Because the errors come from only the library,
+  and not from other dependencies or the library's clients,
+  you can make an exhaustive list of all possible failures.
+  And because these errors are an implementation detail of the library,
+  they're always handled within that library.
+
+- In code that throws only errors that were thrown elsewhere,
+  like a function that takes a closure argument
+  and propagates any errors from that closure.
```

Comment:
I'd like to mention that this is generic code here. We could say "in generic code", or do something like this:

```suggestion
- In code that only propagates errors described by generic parameters, 
  like a function that takes a closure argument
  and propagates any errors from that closure.
```